### PR TITLE
feat: upgrade to NET 8

### PR DESCRIPTION
**Pull request by HBTeun**
Upgrade to NET 8

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [11.0.0]
+
+- Updated to NET 8 SDK container
```
Comment (matijs-toonen) [inline CHANGELOG.md:10]: Also mention that you added support for running .NET 7 inside the build container

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,7 @@
-FROM mcr.microsoft.com/dotnet/sdk:7.0.404-1-bullseye-slim-amd64
+FROM mcr.microsoft.com/dotnet/sdk:8.0
+
+# "install" the dotnet 7 runtime so we can also run the NET 7 tests
```
Comment (matijs-toonen) [inline Dockerfile:3]: 👍

**Review by matijs-toonen — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,7 @@
-FROM mcr.microsoft.com/dotnet/sdk:7.0.404-1-bullseye-slim-amd64
+FROM mcr.microsoft.com/dotnet/sdk:8.0
```
Comment (matijs-toonen) [inline Dockerfile:1]: Should we not use the bullseye slim version anymore?

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,7 @@
...
+
+# "install" the dotnet 7 runtime so we can also run the NET 7 tests
+COPY --from=mcr.microsoft.com/dotnet/sdk:7.0 /usr/share/dotnet/shared /usr/share/dotnet/shared
```
Comment (matijs-toonen) [inline Dockerfile:4]: Should we not use the bullseye slim version anymore?

File: CHANGELOG.md (hunk 1 above)
Comment (HBTeun) [reply CHANGELOG.md]: Did the change

File: Dockerfile (hunk 3 above)
Comment (HBTeun) [reply Dockerfile:1]: bullseye slim is debian 11 based, this one is based on the newest debian 12

File: Dockerfile (hunk 4 above)
Comment (HBTeun) [reply Dockerfile:4]: same as above also it doesn't use the container it just copies over the files and I keep it the same as the FROM step

**Review by Mastermindzh — APPROVED** [review]

File: Dockerfile (hunk 3 above)
Comment (HBTeun) [reply Dockerfile:1]: I thought slim was part of the bullseye name

File: Dockerfile (hunk 4 above)
Comment (HBTeun) [reply Dockerfile:4]: It should be the same as above only for net 7

**Review by Mastermindzh — APPROVED** [review]

**Pull request merged**